### DISCUSSION TITLE
[Brew Absolute Beginner] Proper way of ensuring another cask is installed and then installing a plugin into a folder of the other app?

### DISCUSSION THREAD

**Discussion by thediveo** [Homebrew/discussions#5416] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
/* not yet there, unfortunately */
```

### Output of `brew doctor`

```shell
/* not yet there, unfortunately */
```

### Description of issue

I'm not a macos user/developer (but Linux/Win dev), but would like to improve the installation experience for my macos users. The project is https://github.com/siemens/cshargextcap, that in turn belongs to the https://github.com/siemens/edgeshark project for discovering and capturing container network traffic in container hosts. The intended cask (is this correct?) should:

1. ensure that the [Wireshark cask](https://formulae.brew.sh/cask/wireshark) is installed or gets installed before proceeding.
2. download the CPU architecture-specific plugin binary archive for macos from our Github releases and place the unpacked plugin binary into the host-wide plugins folder of the Wireshark.app.
   - plugin binary might be built from Go sources, is this really necessary?
   - URL scheme handler is unfortunately one of those ugly AppleScript zips with lots of blobs.
4. download the architecture-neutral URL scheme handler .app archive and install it into the place where other .apps get installed into.

Unfortunately, I'm yet unclear about some probably very basic questions in order to get started on writing, testing, and crashing...
- where are the casks hosted, are they always centrally hosted or can I provide the cask directly from our Github releases? From https://docs.brew.sh/Acceptable-Casks I would expect that it would not make sense to go upstream, and instead handle the cask in our repo only. One of the reasons I would expect to be that the binaries are *unsigned*, due to the hassle and complexity of how Apple handles this and our limited resources (this is OpenSource, not a product, after all).
- how do I develop and test my cask locally, that is, how do I tell brew to use a local cask? `brew install --cask ./mycask.rb`?
- can I handle the above mentioned three steps in a single cask, or do I need more casks and/or brews? From reading the documentation I'm not really clear about this, and LLMs just mumble around.

**Comment by p-linnane** [comment, marked as answer]
> where are the casks hosted, are they always centrally hosted or can I provide the cask directly from our Github releases? 

Casks can be hosted at any publicly accessible URL. Many casks use GitHub releases.

> One of the reasons I would expect to be that the binaries are unsigned, due to the hassle and complexity of how Apple handles this and our limited resources (this is OpenSource, not a product, after all).

This is fine if you're hosting it in your own tap, but we won't be able to accept this into Homebrew/cask if the binaries aren't signed.

> how do I develop and test my cask locally, that is, how do I tell brew to use a local cask? `brew install --cask ./mycask.rb`?

`HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>`

> can I handle the above mentioned three steps in a single cask, or do I need more casks and/or brews?

Homebrew installs packages into their own directories. It will not install components into other package's directories.

**Reply by thediveo** [reply]
> Homebrew installs packages into their own directories. It will not install components into other package's directories.

ah, thank you for pointing this out; this is then the show stopper, so it doesn't make sense for me to further follow the brew route.

**Reply by krehel** [reply]
@thediveo - not sure how much it matters to you specifically, but an additional note here is adding plugins to the application bundle does invalidate the signature of the application.  Not familiar enough with Wireshark to comment on alternative location for install or common practice, just more a heads up.

The first screenshot is the standard Wireshark application.  The right side is all green and valid.

The second is with the plugin added per instructions on your repository.  Signature becomes invalidated and Gatekeeper status becomes unknown.

<img width="1509" alt="ws good" src="https://github.com/Homebrew/discussions/assets/39449589/3237ead4-e002-4a96-826e-f9177d113083">
<img width="1509" alt="ws bad" src="https://github.com/Homebrew/discussions/assets/39449589/5f4c311f-90b0-41e6-8534-664153eb0178">

**Reply by thediveo** [reply]
Good point! I looked at the [Wireshark documentation on plugin folders](https://www.wireshark.org/docs/wsug_html_chunked/ChPluginFolders.html):

> If you are running on macOS and Wireshark is installed as an application bundle, the global plugin folder is %APPDIR%/Contents/PlugIns/wireshark, otherwise it’s INSTALLDIR/lib/wireshark/plugins.

Now, I'm not really acquainted with macos here, but is this the location of the whole Wireshark "package"?

Somehow, the Linux package managers seem to be much more modular in this respect, and you can still verify things, if needed.
